crates/bifrost/src/watchdog.rs

```diff
@@ -89,21 +114,46 @@ impl Watchdog {
                 if trim_point == Lsn::INVALID {
                     return Ok(());
                 }
+
+                sender.send((
+                    log_id,
+                    TrimRequest {
+                        requested_trim_point,
+                        actual_trim_point: trim_point,
+                    },
+                ))?;
```

Comment:
This extra round of sending `TrimRequest` through a self channel to then spawn the trim task seems a bit convoluted. Would it be possible to provide a `requested_trim_points: Vec<Lsn>` to this task and once all of them are resolved directly call `trim_chain_if_needed` within the same task? Or were you concerned that looking up the trim point and then trimming might take too long so that pipelining these tasks is beneficial?

Reply:
the only reason i didnt do it is that it would mean we re look up the trim point on every attempt - while this makes it less stale, it delays our enaction of the originally requested trim. in practice its probably fine though.